### ISSUE TITLE
As a traffic events Data Consumer I would like to see all A22 traffic events that are also visible on the A22 web site (and much more)!

### ISSUE THREAD

**Issue by rcavaliere** [noi-techpark/bdp-commons#600]

**Comment by rcavaliere** [comment]
@dulvui I have checked a little bit this morning. I think we need to check the Data Collector. I have seen at least two issues:

- [ ] not all the events types are present, for example I don't see queues, accident or similar. Are there any particular limitations when reading the data from the source?
- [ ] the duration of the event is in some cases strange, i.e. the end_time is before the start_time

It could also be that we have a data quality issue, but we have to investigate this. We need some detailed analysis here by comparing data made available through the A22 API - data stored in the Open Data Hub - data visualized on analytics. In case, let's take some time together for these tests...

**Comment by dulvui** [comment]
@rcavaliere The duration of the events is now shown correctly. It was shown in AM/PM format, instead of 24H format

**Comment by rcavaliere** [comment]
@dulvui yes, this looks OK now. We have here now to understand if we are receiving all data from the A22 web-service. I would in particular check if events of certain categories like queues, accidents or similar, which are shown on the A22 information channels are provided through the web-service or not. More specific: try to make some curl requests to the A22 web-service and check if this is consistent with the data we integrated. If yes, I will contact A22 and better explain this. If no, we need to understand where the issue in the Data Collector is.

**Comment by dulvui** [comment]
@rcavaliere I saw now in the logs, that some events have now a new 'idsottotipoevento' that is not present in the sottotipi csv
https://github.com/noi-techpark/bdp-commons/blob/main/data-collectors/event-a22/src/main/resources/it/bz/noi/a22/events/SottotipiEventi.csv
This is happening since 07.04.2023, so it is not correlated to the events we are currently missing.
I'm still investigating on the missing events.

Here the list of the new 'idsottotipoevento' ids:
```
A22:BrennerLEC_idsottotipoevento:229
A22:BrennerLEC_idsottotipoevento:230
A22:BrennerLEC_idsottotipoevento:233
A22:BrennerLEC_idsottotipoevento:234
A22:BrennerLEC_idsottotipoevento:236
A22:BrennerLEC_idsottotipoevento:237
A22:BrennerLEC_idsottotipoevento:240
A22:BrennerLEC_idsottotipoevento:241
A22:BrennerLEC_idsottotipoevento:244
A22:BrennerLEC_idsottotipoevento:245
A22:BrennerLEC_idsottotipoevento:249
A22:BrennerLEC_idsottotipoevento:250
A22:BrennerLEC_idsottotipoevento:253
A22:BrennerLEC_idsottotipoevento:254
A22:BrennerLEC_idsottotipoevento:256
A22:BrennerLEC_idsottotipoevento:257
A22:BrennerLEC_idsottotipoevento:260
A22:BrennerLEC_idsottotipoevento:264
A22:idtipoevento:26_idsottotipoevento:225
A22:idtipoevento:26_idsottotipoevento:226
```

**Comment by rcavaliere** [comment]
@dulvui yes, I am aware of this and this mainly related to the fact that the variable speed limits of BrennerLEC have been extended to other sections of the A22 and they have prepared new standard events template for them when activated. I will ask A22 to give us their explanation so to update the descriptions in our CSV file

**Comment by dulvui** [comment]
@rcavaliere I saw now that some events have the same start and end date in the interval. The method that checks if a event is valid doesn't allow to have start and end date the same.
I added now +1 second to the end if they are the same on testing, so this problem should not happen anymore. I checked on the writer side and changing it to accept an interval with start = end is not possible.
For now it is live in testing. Next week we can check, if we get now the missing events.

**Comment by rcavaliere** [comment]
@dulvui I am still skeptical that we do not receive or handle all events. Let's try next week to make together some calls to the API of A22, and let's see what we get.

**Comment by rcavaliere** [comment]
Results of the internal session with @dulvui:

- [ ] It seems that we get data only in relation to certain events types. @rcavaliere checks with A22 supplier to verify this
- [ ] We are missing some codes associated to event and subevents types, which we actually map manually since the webservice does not support this. @rcavaliere has requested this to A22
- [ ] We have seen that the Data Collector is not implemented in an optimal way, it seems that it continuously requests a huge amount of historical data each 10 minutes. @dulvui will improve the logic here.
- [ ] The are some edge cases as reported above of events with start date = end date. @dulvui will provide some examples to @rcavaliere and decide again if and how to review the current condition of "valid" event (at present end date should > than start date)

**Comment by rcavaliere** [comment]
Discussion on-going with A22 in order to get ALL event categories (should not be a problem)

**Comment by sseppi** [comment]
@rcavaliere The events should be now into the Open Data Hub, can we close this issue?

**Comment by rcavaliere** [comment]
yes!
